ui/src/message_popup/transformations.jsx

```diff
@@ -20,8 +20,9 @@ export function withIndicator(question) {
   };
 }
 
-export function questionsForIndicator(indicatorId) {
-  const withIndicators = _.compact(allQuestions.map((question) => {
+export function questionsForIndicator(indicatorId, questions) {
+  const questionsList = (questions === undefined) ? allQuestions : questions;
```

Comment:
I'd make this more strict, so that you have to pass in `questions`  That may involve updating a few call sites but makes explicit that this module can't be expected to get `allQuestions` on its own anymore and callers have to handle that.